osdep/atomic.h

```diff
@@ -22,90 +22,10 @@
 #include <inttypes.h>
 #include "config.h"
 
-#if HAVE_STDATOMIC
 #include <stdatomic.h>
 typedef _Atomic float mp_atomic_float;
 typedef _Atomic double mp_atomic_double;
 typedef _Atomic int64_t mp_atomic_int64;
 typedef _Atomic uint64_t mp_atomic_uint64;
```

Comment:
now that those typedefs are not needed, we can just inline standard typedefs, like `atomic_uint_fast64_t` and so on.

Reply:
If there aren't any objection towards this PR then I intended to do that (along with nuking `osdep/atomic.c`).

I'll give a day or two to allow more people to comment before going all in.

Reply:
> like `atomic_uint_fast64_t` and so on

After looking a bit more, it seems that the standard only guarantees these typedef for the `fast` and `least` variants. I.e no `atomic_uint64` is guaranteed.

IME, the `fast` and `least` variants are almost always misused and unnecessary distraction. I'd either keep the `mp_atomic_*` typedefs or just inline `_Atomic uint64_t` instead.

Reply:
The "least" ones exist because of platforms where the exact width ints might not exist (they are optional by the standard). But if that was the case, mpv wouldn't build at all. So it's just useless visual noise.

The "fast" variant also don't really have any actual meaning but has similar issues with useless visual noise. There's also corner cases where the "fast" variant making the integer wider can cause subtle semantic changes. E.g if a platform only supports 32-bit lockless atomic and the "fast" `int32` upgrades it to 64bit then it will silently use locks underneath.

Just typedef-ing or inlining `_Atomic` qualifier on to what we *really* want seems much more straight forward than using `fast/least` stuff just because exact width typedefs weren't available.

Reply:
> E.g if a platform only supports 32-bit lockless atomic and the "fast" int32 upgrades it to 64bit then it will silently use locks underneath.

Isn't this the exact reason `fast` exist for it not to happen and use fast type that is available on the given platform? Eg. it would do the opposite than you said.

Reply:
The standard doesn't really define anything at all about what "fast" means here. So it's entirely up to the implementation to decide what it does/means. There's also no connection between the "fast" types and lockless atomics - as far as the standard goes.

Reply:
Ok, good. I'm fine either way,